encoding/test_encoding.py

```diff
@@ -0,0 +1,242 @@
+            "Content-type: text/html\r\n"
+            f"Last-Modified: {DATE}\r\n"
+            f"Date: {DATE}\r\n"
+            "Server: Deproxy Server\r\n"
+            "Transfer-Encoding: gzip, chunked\r\n"
+            "Content-Length: 33\r\n\r\n"
+            "the body does not actually matter",
+        }
+    ]
+    tempesta = {
+        "config": """
+        server ${server_ip}:8000;
+        """
+    }
+
+    def test(self):
+        self.start_all()
+
+        client = self.get_client("client")
+        server = self.get_server("backend")
+
+        got_response = self.send_req(client)
+        self.assertTrue(
+            client.connection_is_closed(), "The connection has to be closed"
+        )
+        self.assertFalse(got_response, "There should be no response")
+
+
+class TestTEMovedToCE(tester.TempestaTest, CommonUtils):
+    """
+    Ensures that if a response to an `h2` client has
+    `Transfer-Encoding: chunked`, it gets moved to `Content-Encoding`.
+    """
+
+    clients = [
+        {
+            "id": "client",
+            "type": "deproxy_h2",
+            "addr": "${tempesta_ip}",
+            "port": "443",
+            "ssl": True,
+            "ssl_hostname": "localhost",
+        }
+    ]
+    backends = [
+        {
+            "id": "backend",
+            "type": "deproxy",
+            "port": "8000",
+            "response": "static",
+            "response_content":
+            "HTTP/1.1 200 OK\r\n"
+            "Connection: keep-alive\r\n"
+            "Content-type: text/html\r\n"
+            f"Last-Modified: {DATE}\r\n"
+            f"Date: {DATE}\r\n"
+            "Server: Deproxy Server\r\n"
+            "Transfer-Encoding: chunked\r\n\r\n"
+            "the body does not actually matter",
+        }
+    ]
+    tempesta = {
+        "config": """
+        listen 443 proto=h2;
+        server ${server_ip}:8000;
+        tls_certificate ${tempesta_workdir}/tempesta.crt;
+        tls_certificate_key ${tempesta_workdir}/tempesta.key;
+        tls_match_any_server_name;
+        """
+    }
+
+    def test(self):
+        self.start_all()
+
+        client = self.get_client("client")
+        server = self.get_server("backend")
+
+        request = [
+            (":authority", "localhost"),
+            (":path", "/"),
+            (":scheme", "https"),
+            (":method", "GET"),
+        ]
+        client.make_request(request)
+        got_response = client.wait_for_response(timeout=5)
+        response = client.responses[-1] if len(client.responses) else None
+
+        self.assertTrue(got_response, "Got no response")
+        self.assertFalse(
+            "Transfer-Encoding" in response.headers,
+            "The response should not have Transfer-Encoding",
+        )
+        ce = response.headers.get("Content-Encoding", None)
+        self.assertTrue(ce, "The response should have Content-Encoding")
+        self.assertEqual(ce, "chunked", "Wrong Content-Encoding value")
```

Comment:
In the test `TestTEMovedToCE` `Transfer-Encoding: chunked` can't be moved to `Cotent-Encoding` it must be removed from response to client, also tempesta must add correct `Content-Length` header and "dechunk" chunked body.

If you want to test moving encoding from `Transfer-Encoding` to `Content-Encoding`, backend response have to contain `Transfer-Encoding` other than `chunked` and `Content-Length`.